RabbitMQ: Dead letter exchange is always configured with default

When configuring a queue, the dead letter exchange is configured, even though I did not configure it.
This is a problem for me, as I want to connect to an existing RabbitMQ (which I have no control over), where the queue might be already declared without the dead letter exchange.

Error message when trying to connect to the existing RabbitMQ:

```
Caused by: com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=406, reply-text=PRECONDITION_FAILED - inequivalent arg 'x-dead-letter-exchange' for queue 'incoming' in vhost '/': received the value 'DLX' of type 'longstr' but current is none, class-id=50, method-id=10)
```

I understand that `dead-letter-exchange` has a default value, but from my understanding it should only be applicable when `auto-bind-dlq` is set (which I did not set).

When declaring the queue via SmallRye on an empty RabbitMQ I can see in the GUI of RabbitMQ that the `x-dead-letter-exchange` is set to `DLX` (the default value). I think that this related to my problem, but I might be off track.

Using:
- smallrye-reactive-messaging-api 3.13.0
- Rabbit MQ 3.10 from docker image
- Java 11.0.9

This is my configuration
```
rabbitmq-host=localhost
 rabbitmq-port=5672 
rabbitmq-username=guest 
rabbitmq-password=guest  

mp.messaging.incoming.create-incident.connector=smallrye-rabbitmq
 mp.messaging.incoming.create-incident.exchange.name=incoming
mp.messaging.incoming.create-incident.exchange.durable=true 
mp.messaging.incoming.create-incident.exchange.type=topic 
mp.messaging.incoming.create-incident.queue.name=incoming
```

--- a/documentation/src/main/docs/rabbitmq/receiving-messages-from-rabbitmq.md
+++ b/documentation/src/main/docs/rabbitmq/receiving-messages-from-rabbitmq.md
@@ -203,6 +203,15 @@ mp.messaging.incoming.people.queue.declare=true
     name of the exchange; if this is not the case you would need to name the
     exchange explicitly using the `exchange.name` property.
 
+!!!note
+    The connector supports RabbitMQ's "Server-named Queues" feature to create
+    an exclusive, auto-deleting, non-durable and randomly named queue. To
+    enable this feature you set the queue name to exactly `(server.auto)`.
+    Using this name not only enables the server named queue feature but also
+    automatically makes ths queue exclusive, auto-deleting, and non-durable;
+    therefore ignoring any values provided for the `exclusive`, `auto-delete`
+    and `durable` options.
+
 If you want RabbitMQ to create the `people` exchange, queue and binding,
 you need the following configuration:
 

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQClientHelper.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQClientHelper.java
@@ -65,11 +65,15 @@ public class RabbitMQClientHelper {
     static RabbitMQClient getClient(Vertx vertx, RabbitMQConnectorCommonConfiguration config,
             Instance<CredentialsProvider> credentialsProviders) {
         try {
+            String connectionName = String.format("%s (%s)",
+                    config.getChannel(),
+                    config instanceof RabbitMQConnectorIncomingConfiguration ? "Incoming" : "Outgoing");
             String host = config.getHost();
             int port = config.getPort();
             log.brokerConfigured(host, port, config.getChannel());
 
             RabbitMQOptions options = new RabbitMQOptions()
+                    .setConnectionName(connectionName)
                     .setHost(host)
                     .setPort(port)
                     .setSsl(config.getSsl())

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
@@ -217,7 +217,7 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
 
         final Integer connectionCount = ic.getConnectionCount();
         Multi<? extends Message<?>> multi = Multi.createFrom().range(0, connectionCount)
-                .flatMap(ignore -> {
+                .onItem().transformToUniAndMerge(connectionIdx -> {
                     // Create a client
                     final RabbitMQClient client = RabbitMQClientHelper.createClient(this, ic, clientOptions,
                             credentialsProviders);
@@ -229,11 +229,15 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
                                 .subscribe().with(ignored -> promise.complete(), promise::fail);
                     });
                     final ConnectionHolder holder = new ConnectionHolder(client, ic, getVertx());
-                    return holder.getOrEstablishConnection().replaceWith(Tuple2.of(holder, client)).toMulti();
+                    return holder.getOrEstablishConnection()
+                            .invoke(() -> log.connectionEstablished(connectionIdx, ic.getChannel()))
+                            .flatMap(connection -> createConsumer(ic, connection).map(consumer -> Tuple2.of(holder, consumer)));
                 })
-                .flatMap(tuple -> createConsumer(ic, tuple.getItem2()).toMulti()
-                        .map(consumer -> Tuple2.of(tuple.getItem1(), consumer)))
-                .onItem().invoke(connection -> incomingChannelStatus.put(ic.getChannel(), ChannelStatus.CONNECTED))
+                // Wait for all consumers to be created/connected
+                .collect().asList()
+                .onItem().invoke(() -> incomingChannelStatus.put(ic.getChannel(), ChannelStatus.CONNECTED))
+                // Translate all consumers into a merged stream of messages
+                .onItem().transformToMulti(tuples -> Multi.createFrom().iterable(tuples))
                 .flatMap(tuple -> getStreamOfMessages(tuple.getItem2(), tuple.getItem1(), ic, onNack, onAck));
 
         if (Boolean.TRUE.equals(ic.getBroadcast())) {

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
@@ -22,7 +22,6 @@ import javax.enterprise.inject.Any;
 import javax.enterprise.inject.Instance;
 import javax.inject.Inject;
 
-import com.rabbitmq.client.AMQP;
 import org.eclipse.microprofile.config.Config;
 import org.eclipse.microprofile.reactive.messaging.Message;
 import org.eclipse.microprofile.reactive.messaging.spi.Connector;
@@ -33,6 +32,7 @@ import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
 import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
 import org.reactivestreams.Subscription;
 
+import com.rabbitmq.client.AMQP;
 import com.rabbitmq.client.impl.CredentialsProvider;
 
 import io.smallrye.mutiny.Multi;
@@ -176,7 +176,6 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
         TracingUtils.initialise();
     }
 
-    @SuppressWarnings({ "rawtypes", "unchecked" })
     private Multi<? extends Message<?>> getStreamOfMessages(RabbitMQConsumer receiver,
             ConnectionHolder holder,
             RabbitMQConnectorIncomingConfiguration ic,
@@ -190,7 +189,7 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
         log.receiverListeningAddress(queueName);
 
         // The processor is used to inject AMQP Connection failure in the stream and trigger a retry.
-        BroadcastProcessor processor = BroadcastProcessor.create();
+        BroadcastProcessor<Message<?>> processor = BroadcastProcessor.create();
         receiver.exceptionHandler(t -> {
             log.receiverError(t);
             processor.onError(t);
@@ -198,7 +197,7 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
 
         return Multi.createFrom().deferred(
                 () -> {
-                    Multi<? extends Message<?>> stream = receiver.toMulti()
+                    Multi<Message<?>> stream = receiver.toMulti()
                             .map(m -> new IncomingRabbitMQMessage<>(m, holder, isTracingEnabled, onNack, onAck,
                                     contentTypeOverride))
                             .map(m -> isTracingEnabled ? TracingUtils.addIncomingTrace(m, queueName, attributeHeaders) : m);
@@ -237,18 +236,19 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
                     // Create a client
                     final RabbitMQClient client = RabbitMQClientHelper.createClient(this, ic, clientOptions,
                             credentialsProviders);
+                    client.getDelegate().addConnectionEstablishedCallback(promise -> {
+                        // Ensure we create the queues (and exchanges) from which messages will be read
+                        Uni.createFrom().nullItem()
+                                .onItem().call(() -> establishQueue(client, ic))
+                                .onItem().call(() -> establishDLQ(client, ic))
+                                .subscribe().with(ignored -> promise.complete(), promise::fail);
+                    });
                     final ConnectionHolder holder = new ConnectionHolder(client, ic, getVertx());
-                    // Ensure we set the queue up
-                    Uni<RabbitMQClient> uniQueue = holder.getOrEstablishConnection()
-                            // Once connected, ensure we create the queue from which messages are to be read
-                            .onItem().call(connection -> establishQueue(connection, ic))
-                            // If directed to do so, create a DLQ
-                            .onItem().call(connection -> establishDLQ(connection, ic))
-                            .onItem().invoke(connection -> incomingChannelStatus.put(ic.getChannel(), ChannelStatus.CONNECTED));
-                    return uniQueue.replaceWith(Tuple2.of(holder, client)).toMulti();
+                    return holder.getOrEstablishConnection().replaceWith(Tuple2.of(holder, client)).toMulti();
                 })
                 .flatMap(tuple -> createConsumer(ic, tuple.getItem2()).toMulti()
                         .map(consumer -> Tuple2.of(tuple.getItem1(), consumer)))
+                .onItem().invoke(connection -> incomingChannelStatus.put(ic.getChannel(), ChannelStatus.CONNECTED))
                 .flatMap(tuple -> getStreamOfMessages(tuple.getItem2(), tuple.getItem1(), ic, onNack, onAck))
                 .plug(m -> {
                     if (attempts > 0) {
@@ -338,12 +338,15 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
 
         // Create a client
         final RabbitMQClient client = RabbitMQClientHelper.createClient(this, oc, clientOptions, credentialsProviders);
+        client.getDelegate().addConnectionEstablishedCallback(promise -> {
+            // Ensure we create the exchange to which messages are to be sent
+            Uni.createFrom().nullItem()
+                    .onItem().call(ignored -> establishExchange(client, oc))
+                    .subscribe().with((ignored) -> promise.complete(), promise::fail);
+        });
 
         final ConnectionHolder holder = new ConnectionHolder(client, oc, getVertx());
         final Uni<RabbitMQPublisher> getSender = holder.getOrEstablishConnection()
-                // Once connected, ensure we create the exchange to which messages are to be sent
-                .onItem().call(connection -> establishExchange(connection, oc))
-                // Once exchange exists, create ourselves a publisher
                 .onItem().transformToUni(connection -> Uni.createFrom().item(RabbitMQPublisher.create(getVertx(), connection,
                         new RabbitMQPublisherOptions()
                                 .setReconnectAttempts(oc.getReconnectAttempts())
@@ -482,7 +485,7 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
                                 declare = client.queueDeclare(serverQueueName, false, true, true);
                             } else {
                                 declare = client.queueDeclare(serverQueueName, ic.getQueueDurable(),
-                                                              ic.getQueueExclusive(), ic.getQueueAutoDelete(), queueArgs);
+                                        ic.getQueueExclusive(), ic.getQueueAutoDelete(), queueArgs);
                             }
 
                             return declare

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
@@ -464,8 +464,6 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
                                     .onItem().invoke(() -> log.queueEstablished(queueName))
                                     .onFailure().invoke(ex -> log.unableToEstablishQueue(queueName, ex))
                                     .onItem().transformToMulti(v -> establishBindings(client, ic))
-                                    // What follows is just to "rejoin" the Multi stream into a single Uni emitting the queue name
-                                    .onCompletion().invoke(() -> Multi.createFrom().item("ignore"))
                                     .onItem().ignoreAsUni().onItem().transform(v -> queueName);
                         });
     }

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
@@ -22,6 +22,7 @@ import javax.enterprise.inject.Any;
 import javax.enterprise.inject.Instance;
 import javax.inject.Inject;
 
+import com.rabbitmq.client.AMQP;
 import org.eclipse.microprofile.config.Config;
 import org.eclipse.microprofile.reactive.messaging.Message;
 import org.eclipse.microprofile.reactive.messaging.spi.Connector;
@@ -271,7 +272,7 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
     }
 
     private Uni<RabbitMQConsumer> createConsumer(RabbitMQConnectorIncomingConfiguration ic, RabbitMQClient client) {
-        return client.basicConsumer(ic.getQueueName(), new QueueOptions()
+        return client.basicConsumer(serverQueueName(ic.getQueueName()), new QueueOptions()
                 .setAutoAck(ic.getAutoAcknowledgement())
                 .setMaxInternalQueueSize(ic.getMaxIncomingInternalQueueSize())
                 .setKeepMostRecent(ic.getKeepMostRecent()));
@@ -473,15 +474,25 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
         return establishExchange(client, ic)
                 .onItem().transform(v -> Boolean.TRUE.equals(ic.getQueueDeclare()) ? null : queueName)
                 .onItem().ifNull().switchTo(
-                        () -> client
-                                .queueDeclare(queueName, ic.getQueueDurable(), ic.getQueueExclusive(), ic.getQueueAutoDelete(),
-                                        queueArgs)
-                                .onItem().invoke(() -> log.queueEstablished(queueName))
-                                .onFailure().invoke(ex -> log.unableToEstablishQueue(queueName, ex))
-                                .onItem().transformToMulti(v -> establishBindings(client, ic))
-                                // What follows is just to "rejoin" the Multi stream into a single Uni emitting the queue name
-                                .onCompletion().invoke(() -> Multi.createFrom().item("ignore"))
-                                .onItem().ignoreAsUni().onItem().transform(v -> queueName));
+                        () -> {
+                            String serverQueueName = serverQueueName(queueName);
+
+                            Uni<AMQP.Queue.DeclareOk> declare;
+                            if (serverQueueName.isEmpty()) {
+                                declare = client.queueDeclare(serverQueueName, false, true, true);
+                            } else {
+                                declare = client.queueDeclare(serverQueueName, ic.getQueueDurable(),
+                                                              ic.getQueueExclusive(), ic.getQueueAutoDelete(), queueArgs);
+                            }
+
+                            return declare
+                                    .onItem().invoke(() -> log.queueEstablished(queueName))
+                                    .onFailure().invoke(ex -> log.unableToEstablishQueue(queueName, ex))
+                                    .onItem().transformToMulti(v -> establishBindings(client, ic))
+                                    // What follows is just to "rejoin" the Multi stream into a single Uni emitting the queue name
+                                    .onCompletion().invoke(() -> Multi.createFrom().item("ignore"))
+                                    .onItem().ignoreAsUni().onItem().transform(v -> queueName);
+                        });
     }
 
     /**
@@ -501,7 +512,7 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
                 .map(String::trim).collect(Collectors.toList());
 
         return Multi.createFrom().iterable(routingKeys)
-                .onItem().call(routingKey -> client.queueBind(queueName, exchangeName, routingKey))
+                .onItem().call(routingKey -> client.queueBind(serverQueueName(queueName), exchangeName, routingKey))
                 .onItem().invoke(routingKey -> log.bindingEstablished(queueName, exchangeName, routingKey))
                 .onFailure().invoke(ex -> log.unableToEstablishBinding(queueName, exchangeName, ex));
     }
@@ -525,6 +536,13 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
     private RabbitMQAckHandler createAckHandler(RabbitMQConnectorIncomingConfiguration ic) {
         return (Boolean.TRUE.equals(ic.getAutoAcknowledgement())) ? new RabbitMQAutoAck(ic.getChannel())
                 : new RabbitMQAck(ic.getChannel());
+    }
+
+    private String serverQueueName(String name) {
+        if (name.equals("(server.auto)")) {
+            return "";
+        }
+        return name;
     }
 
     public void reportIncomingFailure(String channel, Throwable reason) {

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
@@ -37,7 +37,6 @@ import com.rabbitmq.client.impl.CredentialsProvider;
 
 import io.smallrye.mutiny.Multi;
 import io.smallrye.mutiny.Uni;
-import io.smallrye.mutiny.operators.multi.processors.BroadcastProcessor;
 import io.smallrye.mutiny.tuples.Tuple2;
 import io.smallrye.reactive.messaging.annotations.ConnectorAttribute;
 import io.smallrye.reactive.messaging.health.HealthReport;
@@ -188,21 +187,9 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
                 .map(String::trim).collect(Collectors.toList());
         log.receiverListeningAddress(queueName);
 
-        // The processor is used to inject AMQP Connection failure in the stream and trigger a retry.
-        BroadcastProcessor<Message<?>> processor = BroadcastProcessor.create();
-        receiver.exceptionHandler(t -> {
-            log.receiverError(t);
-            processor.onError(t);
-        });
-
-        return Multi.createFrom().deferred(
-                () -> {
-                    Multi<Message<?>> stream = receiver.toMulti()
-                            .map(m -> new IncomingRabbitMQMessage<>(m, holder, isTracingEnabled, onNack, onAck,
-                                    contentTypeOverride))
-                            .map(m -> isTracingEnabled ? TracingUtils.addIncomingTrace(m, queueName, attributeHeaders) : m);
-                    return Multi.createBy().merging().streams(stream, processor);
-                });
+        return receiver.toMulti()
+                .map(m -> new IncomingRabbitMQMessage<>(m, holder, isTracingEnabled, onNack, onAck, contentTypeOverride))
+                .map(m -> isTracingEnabled ? TracingUtils.addIncomingTrace(m, queueName, attributeHeaders) : m);
     }
 
     /**
@@ -227,8 +214,6 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
 
         final RabbitMQFailureHandler onNack = createFailureHandler(ic);
         final RabbitMQAckHandler onAck = createAckHandler(ic);
-        final Integer interval = ic.getReconnectInterval();
-        final Integer attempts = ic.getReconnectAttempts();
 
         final Integer connectionCount = ic.getConnectionCount();
         Multi<? extends Message<?>> multi = Multi.createFrom().range(0, connectionCount)
@@ -249,20 +234,7 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
                 .flatMap(tuple -> createConsumer(ic, tuple.getItem2()).toMulti()
                         .map(consumer -> Tuple2.of(tuple.getItem1(), consumer)))
                 .onItem().invoke(connection -> incomingChannelStatus.put(ic.getChannel(), ChannelStatus.CONNECTED))
-                .flatMap(tuple -> getStreamOfMessages(tuple.getItem2(), tuple.getItem1(), ic, onNack, onAck))
-                .plug(m -> {
-                    if (attempts > 0) {
-                        return m
-                                // Retry on failure.
-                                .onFailure().invoke(log::retrieveMessagesRetrying)
-                                .onFailure().retry().withBackOff(ofSeconds(1), ofSeconds(interval)).atMost(attempts)
-                                .onFailure().invoke(t -> {
-                                    incomingChannelStatus.put(ic.getChannel(), ChannelStatus.NOT_CONNECTED);
-                                    log.retrieveMessagesNoMoreRetrying(t);
-                                });
-                    }
-                    return m;
-                });
+                .flatMap(tuple -> getStreamOfMessages(tuple.getItem2(), tuple.getItem1(), ic, onNack, onAck));
 
         if (Boolean.TRUE.equals(ic.getBroadcast())) {
             multi = multi.broadcast().toAllSubscribers();

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
@@ -438,8 +438,10 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
 
         // Declare the queue (and its binding(s) to the exchange, and TTL) if we have been asked to do so
         final JsonObject queueArgs = new JsonObject();
-        queueArgs.put("x-dead-letter-exchange", ic.getDeadLetterExchange());
-        queueArgs.put("x-dead-letter-routing-key", ic.getDeadLetterRoutingKey().orElse(queueName));
+        if (ic.getAutoBindDlq()) {
+            queueArgs.put("x-dead-letter-exchange", ic.getDeadLetterExchange());
+            queueArgs.put("x-dead-letter-routing-key", ic.getDeadLetterRoutingKey().orElse(queueName));
+        }
         ic.getQueueSingleActiveConsumer().ifPresent(sac -> queueArgs.put("x-single-active-consumer", sac));
 
         ic.getQueueTtl().ifPresent(queueTtl -> {

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/i18n/RabbitMQLogging.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/i18n/RabbitMQLogging.java
@@ -2,6 +2,7 @@ package io.smallrye.reactive.messaging.rabbitmq.i18n;
 
 import org.jboss.logging.BasicLogger;
 import org.jboss.logging.Logger;
+import org.jboss.logging.Logger.Level;
 import org.jboss.logging.annotations.*;
 
 @MessageLogger(projectCode = "SRMSG", length = 5)
@@ -125,4 +126,9 @@ public interface RabbitMQLogging extends BasicLogger {
     @LogMessage(level = Logger.Level.WARN)
     @Message(id = 17038, value = "No valid content_type set, failing back to byte[]. If that's wanted, set the content type to application/octet-stream with \"content-type-override\"")
     void typeConversionFallback();
+
+    @LogMessage(level = Level.DEBUG)
+    @Message(id = 17039, value = "Connection '%d' with RabbitMQ broker established for channel `%s`")
+    void connectionEstablished(int connectionIndex, String channel);
+
 }

--- a/smallrye-reactive-messaging-rabbitmq/src/test/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQTest.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/test/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQTest.java
@@ -153,8 +153,8 @@ class RabbitMQTest extends RabbitMQBrokerTestBase {
         // verify bindings
         final JsonObject queueArguments = queue.getJsonObject("arguments");
         assertThat(queueArguments).isNotNull();
-        assertThat(queueArguments.getString("x-dead-letter-exchange")).isEqualTo("DLX");
-        assertThat(queueArguments.getString("x-dead-letter-routing-key")).isEqualTo(queueName);
+        assertThat(queueArguments.getString("x-dead-letter-exchange")).isNull();
+        assertThat(queueArguments.getString("x-dead-letter-routing-key")).isNull();
         assertThat(queueArguments.getLong("x-message-ttl")).isEqualTo(queueTtl);
         assertThat(queueArguments.getBoolean("x-single-active-consumer")).isEqualTo(true);
 

